app/facades/today_schedule_facade.rb

```diff
@@ -15,16 +15,15 @@ def leave_types
     @leave_types ||= LeaveTypeFinder.call
   end
 
-  def holiday_assignments
-    holiday = leave_types.detect{ |leave_type| leave_type.name.eql?('Holiday') }
-    return [] unless holiday
-    @holiday_assignments ||= todays_assignments.select{ |assignment| holiday.id == assignment.project_id }
-  end
+  def ooo_assignments
+    return [] unless leave_types
+
+    leave_types.inject({}) do |assignments, leave_type|
+      matching_assignments = todays_assignments.select { |assignment| leave_type.id == assignment.project_id }
```

Comment:
Rather than looping over `leave_types`. Could you doing something like:

```ruby
todays_assignments.select { |assignment| leave_type_ids.include?(assignment.project_id) }
```

(NB: Not sure if there's any value in doing this over your solution)